modules/default/updatenotification/node_helper.js

```diff
@@ -54,34 +56,78 @@ module.exports = NodeHelper.create({
 		}
 	},
 
-	resolveRemote: async function (moduleFolder) {
-		let git = this.createGit(moduleFolder);
-		let remotes = await git.getRemotes(true);
+	execShell: async function (command) {
+		let res = { stdout: "", stderr: "" };
+		const { stdout, stderr } = await exec(command);
 
-		if (remotes.length < 1 || remotes[0].name.length < 1) {
-			throw new Error("No valid remote for folder " + moduleFolder);
+		res.stdout = stdout;
+		res.stderr = stderr;
+		return res;
+	},
+
+	isGitRepo: async function (moduleFolder) {
+		let res = await this.execShell("cd " + moduleFolder + " && git remote -v");
+		if (res.stderr) {
+			Log.error("Failed to fetch git data for " + moduleFolder + ": " + res.stderr);
+			return false;
+		} else {
+			return true;
 		}
+	},
 
-		return git;
+	getRepoInfo: async function (repo) {
+		let gitInfo = {
+			module: repo.module,
+			// commits behind:
+			behind: 0,
+			// branch name:
+			current: "",
+			// current hash:
+			hash: "",
+			// remote branch:
+			tracking: ""
+		};
+		let res = await this.execShell("cd " + repo.folder + " && git rev-parse HEAD");
+		if (res.stderr) {
+			Log.error("Failed to get current commit hash for " + repo.module + ": " + res.stderr);
+		}
+		gitInfo.hash = res.stdout;
+		res = await this.execShell("cd " + repo.folder + " && git status -sb");
+		if (res.stderr) {
+			Log.error("Failed to get git status for " + repo.module + ": " + res.stderr);
+			// exit without git status info
+			return;
+		}
+		// get branch and remote
```

Comment:
would be a nice bonus in the commentsto see an example of what the res.stdout is (for better undertsanding what happens)